openipam/hosts/managers.py

```diff
@@ -102,12 +102,13 @@ def by_change_perms(self, user, pk=None, ids_only=False):
             else:
                 return qs
 
-    def by_expiring(self, ids_only=False, omit_guests=False):
+    def by_expiring(self, ids_only=False, omit_guests=False, mac_last_seen=False):
```

Comment:
I think this is probably fine, since it's being used as a flag to determine whether to include the MAC address history in the queryset, not used to do any filtering. That said, I think it should probably be renamed to something that more accurately reflects that function, maybe `include_mac_history`?

Reply:
I think this is probably fine, since it's being used as a flag to determine whether to include the MAC address history in the queryset, not used to do any filtering. That said, I think it should probably be renamed to something that more accurately reflects that function, maybe `include_mac_history`?

Reply:
yeah it is okay; this is just more of a nitpick, since `ids_only` and `omit_guests` perform filtering operations i would expect at first glance by its name that this function finds all expiring hosts with its parameters. 

i wouldn't want to have to filter further by anything returned here. but again, that's just me.

Reply:
Personally, I think the `select_related()` call that the `mac_last_seen` flag enables should probably just be chained with `by_expiring()` when building the queryset (so do something like `Host.objects.by_expiring().select_related("mac_history")` when using it), I think that would fit more with how Django querysets are usually used. I don't see much added value in merging either the `select_related()` call _or_ a filter on MAC history into the `by_expiring()` filter. If we do want a single function that does all that, I'd propose an entirely new one that chains `by_expiring()` with `select_related()` and a `filter()` on the MAC history, which would only need to take the timestamp as an argument.